enlighten/Graph.py

```diff
@@ -492,52 +499,41 @@ def update_roi_regions(self, spec):
+        self.remove_roi_region(spec.roi_region_right)
+
+        if not spec.settings.eeprom.has_horizontal_roi():
+            # log.debug("hiding curtains (no ROI to show)")
+            return
 
-        if self.in_pixels():
-            log.debug("setting bounds in px")
-            spec.roi_region_left.setRegion((0, roi_start))
-            spec.roi_region_right.setRegion((roi_end, spec.settings.eeprom.active_pixels_horizontal))
-        elif self.in_wavelengths():
-            log.debug(f"setting bounds in nm")
-            spectrum_start_nm = utils.pixel_to_wavelength(0, spec.settings.eeprom.wavelength_coeffs)
-            spectrum_end_nm = utils.pixel_to_wavelength(spec.settings.eeprom.active_pixels_horizontal, spec.settings.eeprom.wavelength_coeffs)
+        if not self.vignette_roi:
+            # log.debug("hiding curtains (no VigentteROI object)")
+            return
 
-            roi_start_nm = utils.pixel_to_wavelength(roi_start, spec.settings.eeprom.wavelength_coeffs)
-            roi_end_nm = utils.pixel_to_wavelength(roi_end, spec.settings.eeprom.wavelength_coeffs)
+        if self.vignette_roi.enabled:
+            # log.debug("hiding curtains (VignetteROI.enabled True), meaning finges should be hid")
+            return
 
-            spec.roi_region_left.setRegion((spectrum_start_nm, roi_start_nm))
-            spec.roi_region_right.setRegion((roi_end_nm, spectrum_end_nm))
+        # log.debug("showing curtains, because VignetteROI.enabled False")
 
-        elif self.in_wavenumbers():
-            spectrum_start_nm = utils.pixel_to_wavelength(0, spec.settings.eeprom.wavelength_coeffs)
-            spectrum_end_nm = utils.pixel_to_wavelength(spec.settings.eeprom.active_pixels_horizontal, spec.settings.eeprom.wavelength_coeffs)
-            
-            roi_start_nm = utils.pixel_to_wavelength(roi_start, spec.settings.eeprom.wavelength_coeffs)
-            roi_end_nm = utils.pixel_to_wavelength(roi_end, spec.settings.eeprom.wavelength_coeffs)
+        roi = spec.settings.eeprom.get_horizontal_roi()
+        axis = self.generate_x_axis(vignetted=False)
 
-            spectrum_start_cm = utils.wavelength_to_wavenumber(spec.settings.eeprom.excitation_nm, spectrum_start_nm)
-            spectrum_end_cm = utils.wavelength_to_wavenumber(spec.settings.eeprom.excitation_nm, spectrum_end_nm)
+        log.debug(f"update_roi_regions: roi {roi}, axis {len(axis)} elements")
 
-            roi_start_cm = utils.wavelength_to_wavenumber(spec.settings.eeprom.excitation_nm_float, roi_start_nm)
-            roi_end_cm = utils.wavelength_to_wavenumber(spec.settings.eeprom.excitation_nm_float, roi_end_nm)
+        spec.roi_region_left .setRegion((axis[0],       axis[roi.start]))
+        spec.roi_region_right.setRegion((axis[roi.end], axis[-1]       ))
 
-            spec.roi_region_left.setRegion((-1*spectrum_start_cm, -1*roi_start_cm))
-            spec.roi_region_right.setRegion((-1*roi_end_cm, -1*spectrum_end_cm))
```

Comment:
I never did figure out why you had to do those -1, but the new version is simpler